utils/e2e-tests/bash/tests/base/help.sh

```diff
@@ -3,23 +3,44 @@ set -euo pipefail
 
 SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
 
-# This test ensures we have an adequate and working help command.
-# It also serves as a snapshot test for the help text, so we can track when
-# the help text changes.
-
-OUTPUT="$("$HUMANODE_PEER_PATH" --help)"
-TEMPLATE="$(cat "$SCRIPT_DIR/../../fixtures/help.stdout.txt")"
-
-DIFF_CMD_ARGS=(
-  -u
-  -b
-  -I '^humanode-peer {{sha}}$'        # template
-  -I '^humanode-peer [0-9a-f]\{40\}$' # expected
-)
-
-if ! DIFF="$(diff "${DIFF_CMD_ARGS[@]}" <(printf '%s' "$TEMPLATE") <(printf '%s' "$OUTPUT"))"; then
-  printf "Output did not match:\n%s\n" "$DIFF"
-  exit 1
-fi
+# This test ensures we have an adequate and working help command and subcommands.
+# It also serves as a snapshot test for the help texts, so we can track when
+# the help texts change.
+
+compare() {
+  local COMMAND="${@:-""}"
+
+  # Replace commands spaces by dots to prepare fixture filename.
+  FIXTURE_FILENAME="help."${COMMAND// /.}".stdout.txt"
+  # Avoid having double dots.
+  FIXTURE_FILENAME=${FIXTURE_FILENAME//../.}
+
+  OUTPUT="$("$HUMANODE_PEER_PATH" $COMMAND --help)"
+  TEMPLATE="$(cat "$SCRIPT_DIR/../../fixtures/help-output/$FIXTURE_FILENAME")"
+
+  DIFF_CMD_ARGS=(
+    -u
+    -b
+    -I '^humanode-peer {{sha}}$'        # template
+    -I '^humanode-peer [0-9a-f]\{40\}$' # expected
+  )
+
+  if ! DIFF="$(diff "${DIFF_CMD_ARGS[@]}" <(printf '%s' "$TEMPLATE") <(printf '%s' "$OUTPUT"))"; then
+    printf "Output did not match:\n%s\n" "$DIFF"
+    exit 1
+  fi
+
+  local SUBCOMMANDS=($(awk '/Commands:/,/Options:/{if(/Commands:|Options:/) next; print}' <(printf '%s' "$OUTPUT") | awk '{if ($1) print $1}'))
```

Comment:
Why the `awk` command chaining here?

It seems to me like the `awk` invocation here would look better if we extract it into a separate fn.

Same applied to the `if`/`diff` clause.

I'd refactor this a bit further to simplify the subsequent maintenance.

Reply:
Also, there are a ton of shellcheck warnings:
<img width="1179" alt="image" src="https://github.com/humanode-network/humanode/assets/265507/bb968f93-1fa5-4729-889b-f16980ae9ce0">

Reply:
Refactored a lot of stuff, but still left the chained `awk` invocations as is for you to look into.

Reply:
Thanks for the improvements!

The `awk` usage has been refactored - https://github.com/humanode-network/humanode/pull/1034/commits/66cfa4912f927af83ab4796e80906785159978b6.